skl2onnx/common/_container.py

```diff
@@ -95,17 +95,52 @@ def _build_options(model, defined_options, default_values, allowed_options):
 _apply_operation_specific = _get_operation_list()
 
 
-class RawModelContainerNode(object):
+class _WhiteBlackContainer:
+
+    def __init__(self, white_op=None, black_op=None):
+        self._white_op = white_op
+        self._black_op = black_op
+
+    def is_allowed(self, node_type):
+        """
+        Tells if a node is white listed or not black listed.
+        """
+        if isinstance(node_type, (list, tuple, set)):
+            return all(map(self.is_allowed, node_type))
+        try:
+            self.check_white_black_list(node_type)
+            return True
+        except RuntimeError:
+            return False
+
+    def check_white_black_list(self, node_type):
+        """
+        Checks a node type is allowed according to white
+        and black lists.
+        """
+        if self._white_op:
```

Comment:
You could merge the two ifs into one.

Reply:
I thought that it would be better for the user to know which list is involved in the error message.